tool/lib/envutil.rb

```diff
@@ -165,6 +165,11 @@ def invoke_ruby(args, stdin_data = "", capture_stdout = false, capture_stderr =
     }
 
     args = [args] if args.kind_of?(String)
+    # use the same parser as current ruby
+    if args.none? { |arg| arg.to_s =~ /--parser=/ }
+      current_parser = RUBY_DESCRIPTION =~ /prism/i ? "prism" : "parse.y"
+      args << "--parser=#{current_parser}"
+    end
```

Comment:
I need these lines because prism is the current parser now (it wasn't when I made my initial commit). So now it only used prism parser and compiler in a subprocess without this. Now, when parse.y is tested it will use parse.y parser/compiler in subprocess too.